pkg/feature/templates/servicemesh/kserve/kserve-predictor-authorizationpolicy.tmpl

```diff
@@ -0,0 +1,18 @@
+apiVersion: security.istio.io/v1beta1
+kind: AuthorizationPolicy
```

Comment:
@retocode @skonto So, this `AuthorizationPolicy` will allow to protect `InferenceServices` behind auth (per user desire). The pod(s) of the `ISVC` will be protected, with the exception of the two paths mentioned below.

Would this be OK for KNative? I saw the queue-proxy container in the pod and was wondering if there would be any issues with it, or something else. On my naive testings, everything worked fine, but I wanted to check with you about it.

Reply:
Is this only going to run on the main containerPorts traffic (default 8080) or also on the other ones?

For example we also have traffic on:
- admin port 8022
- http2 on 8013
- metrics ports 9090/9091
- /debug/* (if you enable profiling)

Anything I've forgotten @skonto?

Reply:
This would be intercepting everything.

So, I would infer that the default port of 8080 would be used only if the main container does not specify a port, right? From KNative upstream docs, I see that you can set a `containerPort` in the PodSpec... May that user-defined port be used transparently?? However, I see you mention 8013 for http2. Are apps required to use that for http2??? Just asking for my knowledge.

In any case, if the user can choose the port, that would be the motivation for the catch-all.

AFAIK, we can whitelist ports -- @aslakknutsen may confirm. That said, this is cluster-wide config. So, to minimize blackholes, we want to protect everything but ports/paths that would **always** work for infrastructure means.

Reply:
The ports I mentioned are on the Queue-Proxy sidecar. Knative will call the ports depending on the traffic type. So the actual user container will run as 8080 (or as defined by the user), but we'll send traffic to Queue-Proxy ports first. But if the rules are not port-related, we need to focus on the list in https://github.com/opendatahub-io/opendatahub-operator/pull/784#discussion_r1458086265

Reply:
> but we'll send traffic to Queue-Proxy ports first.

You mean that traffic will arrive first to queue-proxy's port 8080 for HTTP1, or 8013 for HTTP2; and later is forwarded to the user container to whatever port it is defined in its spec? Asking just for my knowledge.

I just updated the `AuthorizationPolicy` (as noted in my other comment). Please, check if that would be OK for you :-)

Reply:
> You mean that traffic will arrive first to queue-proxy's port 8080 for HTTP1, or 8013 for HTTP2; and later is forwarded to the user container to whatever port it is defined in its spec? Asking just for my knowledge.

Exactly.

Reply:
> Anything I've forgotten @skonto?

That is fine. To add some details.
```
        // BackendHTTPPort is the backend, i.e. `targetPort` that we setup for HTTP/1 services.
	BackendHTTPPort = 8012

	// BackendHTTP2Port is the backend, i.e. `targetPort` that we setup for HTTP/2 services.
	BackendHTTP2Port = 8013

	// BackendHTTPSPort is the backend. i.e. `targetPort` that we setup for HTTPS services.
	BackendHTTPSPort = 8112

	// QueueAdminPort specifies the port number for
	// health check and lifecycle hooks for queue-proxy.
	QueueAdminPort = 8022

	// AutoscalingQueueMetricsPort specifies the port number for metrics emitted
	// by queue-proxy for autoscaler.
	AutoscalingQueueMetricsPort = 9090

	// UserQueueMetricsPort specifies the port number for metrics emitted
	// by queue-proxy for end user.
	UserQueueMetricsPort = 9091
```

BackendHTTPSPort is not being used with Istio, it is meant for internal encryption with Kourier.

For profiling if enabled:
```
	// ProfilingPort specifies the default port where profiling data is available when profiling is enabled
	ProfilingPort = 8008
```